frontend/src/components/PasswordChange.jsx

```diff
@@ -0,0 +1,378 @@
+    } else if (pwd.length < 8) {
+      setPasswordErrorMessage("Password must be at least 8 characters long");
+    } else if (pwd.search(/[a-z]/i) < 0) {
+      setPasswordErrorMessage("Password must contain at least one letter");
+    } else if (pwd.search(/[A-Z]/) < 0) {
+      setPasswordErrorMessage(
+        "Password must contain at least one capital letter"
+      );
+    } else if (pwd.search(/[0-9]/) < 0) {
+      setPasswordErrorMessage("Password must contain at least one number");
+    } else if (pwd.search(/[ !@#$%^&\*-\._,;ºª\\/()~?¿¡:="·<>{}[\]+]/) < 0) {
+      setPasswordErrorMessage(
+        "Password must contain at least one special character"
+      );
+    } else {
+      setPasswordErrorMessage("");
+      setPasswordValid(true);
+    }
+
+    setPasswordValidated(true);
+  };
+
+  const checkPasswordsMatch = (pwd = "", confirmPwd = "") => {
+    setPasswordsMatch(false);
+    if (pwd.length === 0) {
+      setConfirmPasswordErrorMessage("This field is required");
+    } else if (pwd !== confirmPwd) {
+      setConfirmPasswordErrorMessage("Passwords do not match");
+    } else {
+      setConfirmPasswordErrorMessage("");
+      setPasswordsMatch(true);
+    }
+
+    setConfirmPasswordValidated(true);
+  };
+
+  const isVerificationCodeValid = (code = "") => {
+    if (code.length === 0) {
+      setVerificationCodeErrorMessage("This field is required");
+      setVerificationCodeValid(false);
+    } else if (code.search(/[0-9]/) < 0) {
+      setVerificationCodeErrorMessage("Verification code must be a number");
+      setVerificationCodeValid(false);
+    } else if (code.length !== 6) {
+      setVerificationCodeErrorMessage("Verification code must be 6 digits");
+      setVerificationCodeValid(false);
+    } else {
+      setVerificationCodeErrorMessage("");
+      setVerificationCodeValid(true);
+    }
+    setVerificationCodeValidated(true);
+  };
+
+  const requestPasswordChange = async () => {
+    try {
+      const response = await fetch(
+        "http://0.0.0.0:8000/user/password_recovery/"
+          .concat(email)
+          .concat("?verification_code=")
+          .concat(verificationCode),
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ password: password }),
+        }
+      );
+      if (!response.ok) {
+        setVerificationCodeValid(false);
+        const data = await response.json();
+        if (data && data.detail) {
+          setUpdateConfirmationMessage(data.detail);
+        } else {
+          setUpdateConfirmationMessage(
+            "Could not check verification code. Please try again later."
+          );
+          setShowUpdateConfirmation(true);
+        }
+      } else {
+        const data = await response.json();
+        setVerificationCodeValid(data.status);
+        if (!data.status) {
+          setVerificationCodeErrorMessage("Verification code is incorrect.");
```

Comment:
Good security measures, making it necessary to enter a verification code to change the password.